network.go

```diff
@@ -176,7 +180,22 @@ func (n *network) Delete() error {
 		n.ctrlr.Unlock()
 		return &ActiveEndpointsError{name: n.name, id: string(n.id)}
 	}
+	n.ctrlr.Unlock()
+
+	if err = n.deleteNetwork(); err != nil {
+		return err
+	}
+
+	if err = n.ctrlr.deleteNetworkFromStore(n); err != nil {
```

Comment:
Again `deleteNetworkFromStore` needs to be done atomically from data store before local driver delete is triggered and if the atomic `deleteNetworkFromStore` fails because of stale index an immediate re-update of network endpoint data needs to happen. Which brings the question since networks and endpoints are handled as two separate tables how is this handled in terms of atomically determining how many endpoints are bound to this network?